Analytics/models/pin_location_data.py

```diff
@@ -0,0 +1,339 @@
+import logging
+import random
+from datetime import datetime, timedelta
+from typing import Union
+
+from sqlalchemy import and_, func
+from sqlalchemy.exc import IntegrityError
+
+from db import db
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class Tracker(db.Model):
+    __tablename__ = 'tracker'
+
+    id = db.Column(db.Text, unique=True, primary_key=True, autoincrement=False, nullable=False)
+    description = db.Column(db.Text, nullable=True)
+    activated_date = db.Column(db.DateTime, default=datetime.now, nullable=False)
+    loc_data = db.relationship("LocationData", back_populates="tracker",
+                               primaryjoin="and_(Tracker.id==LocationData.tracker_id)")
+
+    def __init__(self, tracker_id: str, description: str = "",
+                 activated_date: datetime.timestamp = datetime.utcnow()) -> None:
+        """
+        Instantiate Tracker model
+        :param tracker_id: New Tracker Id
+        :param activated_date: Date Activated
+        """
+        self.id = tracker_id
+        self.activated_date = activated_date
+        self.description = description
+
+    @property
+    def json_with_location(self):
+        """
+        JSON Serialize Objects Attributes and Location data
```

Comment:
Please look at your docstrings descriptions. Docs should be written as if telling the code what to do